src/traces/indicator/defaults.js

```diff
@@ -0,0 +1,162 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var handleDomainDefaults = require('../../plots/domain').defaults;
+var Template = require('../../plot_api/plot_template');
+var handleArrayContainerDefaults = require('../../plots/array_container_defaults');
+var cn = require('./constants.js');
+
+var handleTickValueDefaults = require('../../plots/cartesian/tick_value_defaults');
+var handleTickMarkDefaults = require('../../plots/cartesian/tick_mark_defaults');
+var handleTickLabelDefaults = require('../../plots/cartesian/tick_label_defaults');
+
+function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    handleDomainDefaults(traceOut, layout, coerce);
+
+    // Mode
+    coerce('mode');
+    traceOut._hasNumber = traceOut.mode.indexOf('number') !== -1;
+    traceOut._hasDelta = traceOut.mode.indexOf('delta') !== -1;
+    traceOut._hasGauge = traceOut.mode.indexOf('gauge') !== -1;
+
+    coerce('value');
+    coerce('vmin');
+    coerce('vmax', 1.5 * traceOut.value);
+
+    // Number attributes
+    var auto = [];
```

Comment:
Safer to initialise `auto` as `new Array(2)` here before filling in `auto[0]` and `auto[1]`.

Reply:
Done in e6d8525ab07c56ca10bdbac6762b06d1d57bf830